checkov/secrets/runner.py

```diff
@@ -127,7 +129,17 @@ def run(
                     filter_ignored_paths(root, d_names, excluded_paths)
                     filter_ignored_paths(root, f_names, excluded_paths)
                     for file in f_names:
-                        if file not in PROHIBITED_FILES and f".{file.split('.')[-1]}" in SUPPORTED_FILE_EXTENSIONS:
+                        if runner_filter.secrets_scan_file_type:
+                            if 'all' in runner_filter.secrets_scan_file_type:
+                                if 'Dockerfile' in file or f".{file.split('.')[-1]}" in (ADDED_TO_SECRET_SCAN_FILES_TYPES + SUPPORTED_FILE_EXTENSIONS):
+                                    files_to_scan.append(os.path.join(root, file))
+                            else:
+                                if 'Dockerfile' in runner_filter.secrets_scan_file_type:
+                                    if 'Dockerfile' in file:
+                                        files_to_scan.append(os.path.join(root, file))
+                                if f".{file.split('.')[-1]}" in runner_filter.secrets_scan_file_type:
+                                    files_to_scan.append(os.path.join(root, file))
+                        elif file not in PROHIBITED_FILES and f".{file.split('.')[-1]}" in SUPPORTED_FILE_EXTENSIONS:
                             files_to_scan.append(os.path.join(root, file))
```

Comment:
please also add the Dockerfile mask for the old behaviour not just when used with the new flag